lang/java/reef-io/src/main/java/org/apache/reef/io/data/loading/impl/DistributedDataSetPartition.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.io.data.loading.impl;
+
+import org.apache.commons.lang.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.apache.reef.annotations.Unstable;
+
+/**
+ * POJO that represents a distributed data set partition. Basically, it contains the path where
+ * the data files are located for this partition, and the location where we want
+ * this data to be loaded into.
+ *
+ */
+@Unstable
+public final class DistributedDataSetPartition {
+
+  /**
+   * Constant to specify that the data partition could be loaded into any
+   * location.
+   */
+  public static final String LOAD_INTO_ANY_LOCATION = "/*";
+
+  /**
+   * The path of the distributed data set partition. If we use HDFS, it will be the
+   * hdfs path.
+   */
+  private final String path;
+
+  /**
+   * The location (either a rackName or a nodeName) where we want the data
+   * in this distributed partition to be loaded into.
+   */
+  private final String location;
```

Comment:
Can this contain `*`? Such as `/datacenter1/*`? Or is it assumed to be a prefix into the rackname?